codex/rest/api.nim

```diff
@@ -45,6 +48,47 @@ proc validate(
   {.gcsafe, raises: [Defect].} =
   0
 
+proc formatAddress(address: Option[dn.Address]): string =
+  if address.isSome():
+    return $address.get()
+  return "<none>"
+
+proc formatNode(node: dn.Node): JsonNode =
+  let jobj = %*{
+    "nodeId": $node.id,
+    "peerId": $node.record.data.peerId,
+    "record": $node.record,
+    "address": formatAddress(node.address),
+    "seen": $node.seen
+  }
+  return jobj
+
+proc formatTable(routingTable: rt.RoutingTable): JsonNode =
+  let jarray = newJArray()
+  for bucket in routingTable.buckets:
+    for node in bucket.nodes:
+      jarray.add(formatNode(node))
```

Comment:
If I'm not mistaken, in a fully populated p2p network, this could potentially become a very expensive operation, as the last k-bucket could contain half the network (known to the client).

I understand this is being called from the `debug/info` endpoint, but given that this is a single-threaded client and these operations are not async, this could lock up the node and be a DDOS vector.

Perhaps a solution would be to hide this endpoint behind a compile flag so that it doesn't inadvertently make its way into production (similar was done in https://github.com/codex-storage/nim-codex/pull/393/commits/331884a85d946b7f99bdeac64f85eb25310f56df).

Reply:
Good idea. I added the compile flag. :D

Reply:
What do you mean by screwing up execution in a testnet?

Reply:
Like you said, it could be an expensive operation in a real p2p network. So, if we want to be able to do something like this for inspecting a future test-net, then maybe we can come up with a way to implement it that doesn't block threads (so much).

Reply:
Hmm, yes good point. One possible solution might be the result set could be limited and paged perhaps.